src/v/kafka/server/group_tx_tracker_stm.cc

```diff
@@ -0,0 +1,180 @@
+    // Transaction commit/abort ensures the data structures are cleaned
+    // up and bounded in size and all the open transactions are only
+    // in the non evicted part of the log, so nothing to do.
+    return ss::now();
+}
+
+ss::future<iobuf> group_tx_tracker_stm::take_snapshot(model::offset) {
+    return ss::make_ready_future<iobuf>(iobuf());
+}
+
+ss::future<> group_tx_tracker_stm::handle_raft_data(model::record_batch) {
+    // No transactional data, ignore
+    return ss::now();
+}
+
+ss::future<> group_tx_tracker_stm::handle_tx_offsets(
+  model::record_batch_header header, kafka::group_tx::offsets_metadata data) {
+    // in case the fence got truncated, try to start the transaction from
+    // this point on. This is not possible today but may help if delete
+    // retention is implemented for consumer topics.
+    maybe_add_tx_begin_offset(
+      std::move(data.group_id),
+      model::producer_identity{header.producer_id, header.producer_epoch},
+      header.base_offset);
+    return ss::now();
+}
+
+ss::future<> group_tx_tracker_stm::handle_fence_v0(
+  model::record_batch_header header, kafka::group_tx::fence_metadata_v0 fence) {
+    maybe_add_tx_begin_offset(
+      std::move(fence.group_id),
+      model::producer_identity{header.producer_id, header.producer_epoch},
+      header.base_offset);
+    return ss::now();
+}
+
+ss::future<> group_tx_tracker_stm::handle_fence_v1(
+  model::record_batch_header header, kafka::group_tx::fence_metadata_v1 fence) {
+    maybe_add_tx_begin_offset(
+      std::move(fence.group_id),
+      model::producer_identity{header.producer_id, header.producer_epoch},
+      header.base_offset);
+    return ss::now();
+}
+
+ss::future<> group_tx_tracker_stm::handle_fence(
+  model::record_batch_header header, kafka::group_tx::fence_metadata fence) {
+    maybe_add_tx_begin_offset(
+      std::move(fence.group_id),
+      model::producer_identity{header.producer_id, header.producer_epoch},
+      header.base_offset);
+    return ss::now();
+}
+
+ss::future<> group_tx_tracker_stm::handle_abort(
+  model::record_batch_header header, kafka::group_tx::abort_metadata data) {
+    maybe_end_tx(
+      std::move(data.group_id),
+      model::producer_identity{header.producer_id, header.producer_epoch});
+    return ss::now();
+}
+
+ss::future<> group_tx_tracker_stm::handle_commit(
+  model::record_batch_header header, kafka::group_tx::commit_metadata data) {
+    auto pid = model::producer_identity{
+      header.producer_id, header.producer_epoch};
+    maybe_end_tx(std::move(data.group_id), pid);
+    return ss::now();
+}
+
+ss::future<> group_tx_tracker_stm::handle_version_fence(
+  features::feature_table::version_fence) {
+    // ignore
+    return ss::now();
+}
+
+bool group_tx_tracker_stm_factory::is_applicable_for(
+  const storage::ntp_config& config) const {
+    auto nt = model::topic_namespace{config.ntp().ns, config.ntp().tp.topic};
```

Comment:
here we copy topic and namespace for each single batch, maybe we can directly compare topic and namespace or use `model::topic_namespace_view`